oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/user/util/UserUtil.java

```diff
@@ -39,13 +48,43 @@
  */
 public final class UserUtil implements UserConstants {
 
+    private static final Logger log = LoggerFactory.getLogger(UserUtil.class);
+
     private UserUtil() {
     }
 
     public static boolean isAdmin(@NotNull ConfigurationParameters parameters, @NotNull String userId) {
         return getAdminId(parameters).equals(userId);
     }
 
+    public static boolean isMemberOfAnAdministratorGroup(@NotNull Authorizable authorizable, @NotNull ConfigurationParameters parameters) {
```

Comment:
i am not convinced that this method should be part of the public API because it is only evaluating an configuration option related to impersonation, which can be kept internally.... the only place where the configuration option needs to be evaluated is in "ImpersonationImpl" -> move it there or in the non-exported user-utility class in org.apache.jackrabbit.oak.security.user (in oak-core)

so, no need to change the public API and the package version.

also, and this is the most important piece:

the reason why ImpersonationImpl today does NOT support impersonation being granted for groups today is the fact that I don't want group-membership to be evaluated in the 'Impersonation.allows' method. that was the reason for the comment you removed :). because that neither scales nor performs.

So, instead this method should not be limited to any kind of group operation but instead, I would like to have a simple principal-name comparison as follows:

while looping over the set of principal in the subject passed to 'Impersonation.allows', simply test if any of the principal-name matches any of the values present in the configuration. done.

this means:
- not limiting it to group principals
- not making any attempt to evaluate group membership

as long as the principal is present in the subject we are good an no costy evaluation is need. 
if you believe limiting this to group principals would be sensible we can have a simple check for matching name + principal being an instanceof GroupPrincipal.

long story short:
move it to oak-core and simplify